libraries/SRV_Channel/SRV_Channel.cpp

```diff
@@ -170,6 +170,20 @@ float SRV_Channel::get_output_norm(void)
     return ret;
 }
 
+float SRV_Channel::get_output_norm_from_pwm(uint16_t pwm)
+{
+    float ret;
+    if (pwm < servo_trim) {
+        ret = (float)(pwm - servo_trim) / (float)(servo_trim - servo_min);
+    } else if (pwm > servo_trim) {
+        ret = (float)(pwm - servo_trim) / (float)(servo_max  - servo_trim);
+    } else {
+        ret = 0;
+    }
+    
```

Comment:
It does not need reverse as PWM was already reversed when it became PWM...
get_output_norm was not by me and I do not use it due to same reason.
ESC scaling in HAL is legacy I believe, as it uses same ranges for all ESCs that might be present. This one is more agile.